File: Algorithms/Warmup/TimeConversion.java
package Warmup;

import java.util.Scanner;
/**
 *	Algorithms -> Warmup -> Time Conversion
 *	Easy (Super Easy)
 */
public class TimeConversion {

	public static void main(String[] args) {
		Scanner in = new Scanner(System.in);
        String time = in.next();
        in.close();
        String r = convertTime(time);
        System.out.println(r);
	}
	
	public static String convertTime(String time) 
	{
		String substring = time.substring(Math.max(time.length() - 2, 0));
        int hours = 0;
        if(substring.equals("PM")) {
            hours = Integer.parseInt(time.substring(0,2));
            if(hours != 12) {
           	 hours += 12;
            }
            time = hours + time.substring(2,  time.length() - 2);
            return time;
        } else {
            hours = Integer.parseInt(time.substring(0,2));
            if(hours == 12) {
                time = "00" + time.substring(2,  time.length() - 2);   
            } else {
                time = time.substring(0, time.length() - 2);    
            }
            return time;
        }
	}
}

File: Algorithms/Warmup/TimeConversion.c
#include <math.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>
#include <assert.h>
#include <limits.h>
#include <stdbool.h>

/**
 *	Algorithms -> Warmup -> Time Conversion
 *	Easy
 */
int main(){
	char timestamp[11] = "\0\0\0\0\0\0\0\0\0\0\0";
	int hr=0;

	scanf("%s", timestamp);

	if(timestamp[8] == 'P'){
		hr = 10*(timestamp[0]-'0')+(timestamp[1]-'0');
	    if(hr < 12) hr += 12;
	} else{
		hr = 10*(timestamp[0]-'0')+(timestamp[1]-'0');
	    if(hr == 12) hr = 0;
	}

	timestamp[0] = hr/10 + '0';
	timestamp[1] = hr%10 + '0';
	timestamp[8] = '\0';
	timestamp[9] = '\0';

	printf("%s", timestamp);
	return 0;
}
